### PR TITLE
Swe/display quote

### LINKED ISSUE #577: Ability to view approved quote
Button or link somewhere on the project details page or header just for projects with approved quotes (active, completed, etc).

### PR DESCRIPTION
**Pull request by sprucely**
Resolves #577

### CODE REVIEW

**Review by jSylvestre — CHANGES_REQUESTED** [review]
You can't use the quote/get to view an approved quote.
It is checking approved == null or something like that.
Needed for an unapproved quote (maybe)

File: Harvest.Web/Controllers/Api/QuoteController.cs [hunk 1]
```diff
@@ -39,6 +39,22 @@ public async Task<ActionResult> Get(int projectId)
             return Ok(model);
         }
 
+        // Get info on the project as well as in-progess quote if it exists
+        [HttpGet]
+        public async Task<ActionResult> GetApproved(int projectId)
+        {
+            var project = await _dbContext.Projects
+                .Include(p => p.Quote)
```
Comment (jSylvestre) [inline Harvest.Web/Controllers/Api/QuoteController.cs:47]: @sprucely FYI, this is currently null. This PR will start populating it:
https://github.com/ucdavis/Harvest/pull/591

File: Test/TestsControllers/TestsApiControllers/QuoteControllerTests.cs [hunk 2]
```diff
@@ -70,6 +70,10 @@ public void TestControllerMethodAttributes()
             authAttribute.ElementAt(0).Policy.ShouldBe(AccessCodes.SupervisorAccess);
             ControllerReflection.MethodExpectedAttribute<HttpPostAttribute>("Save", countAdjustment + 3);
             ControllerReflection.MethodExpectedAttribute<AsyncStateMachineAttribute>("Save", countAdjustment + 3);
+
+            //4
+            ControllerReflection.MethodExpectedAttribute<HttpGetAttribute>("GetApproved", countAdjustment + 2);
+            ControllerReflection.MethodExpectedAttribute<AsyncStateMachineAttribute>("GetApproved", countAdjustment + 2);
```
Comment (jSylvestre) [inline Test/TestsControllers/TestsApiControllers/QuoteControllerTests.cs:76]: Need to add a test for the details too.

**Review by jSylvestre — APPROVED** [review]
A couple comments

File: Harvest.Web/Controllers/Api/QuoteController.cs (hunk 1 above)
Comment (sprucely) [reply Harvest.Web/Controllers/Api/QuoteController.cs:47]: Wasn't null in my tests against local db.

File: Test/TestsControllers/TestsApiControllers/QuoteControllerTests.cs (hunk 2 above)
Comment (sprucely) [reply Test/TestsControllers/TestsApiControllers/QuoteControllerTests.cs:76]: I'm just going off the established pattern. Where are the detail tests? I just see a bunch of ControllerReflection boilerplate.

Comment (jSylvestre) [reply Test/TestsControllers/TestsApiControllers/QuoteControllerTests.cs:76]: Details is just a get, with no async, so would look like this:
`//5
            ControllerReflection.MethodExpectedAttribute<HttpGetAttribute>("Details", 1);`

Comment (sprucely) [reply Test/TestsControllers/TestsApiControllers/QuoteControllerTests.cs:76]: 🦤 I completely forgot about the Details method. Thought you were asking for a functional test.

**Pull request merged**